File: ios/Classes/Serializer/ToggleStyleSettingsSerializer.swift
import UsercentricsUI

extension ToggleStyleSettings {
    init?(from value: Any?) {
        guard
            let dict = value as? Dictionary<String, Any>
        else { return nil }

        self.init(activeBackgroundColor: UIColor(unsafeHex: dict["activeBackgroundColor"] as? String),
                  inactiveBackgroundColor: UIColor(unsafeHex: dict["inactiveBackgroundColor"] as? String),
                  disabledBackgroundColor: UIColor(unsafeHex: dict["disabledBackgroundColor"] as? String),
                  activeThumbColor: UIColor(unsafeHex: dict["activeThumbColor"] as? String),
                  inactiveThumbColor: UIColor(unsafeHex: dict["inactiveThumbColor"] as? String),
                  disabledThumbColor: UIColor(unsafeHex: dict["disabledThumbColor"] as? String))
    }
}

File: android/src/main/kotlin/com/usercentrics/sdk/flutter/serializer/ToggleStyleSettingsSerializer.kt
package com.usercentrics.sdk.flutter.serializer

import com.usercentrics.sdk.ToggleStyleSettings

internal fun Any?.deserializeToggleStyleSettings(): ToggleStyleSettings? {
    if (this !is Map<*, *>) {
        return null
    }
    return ToggleStyleSettings(
        activeBackgroundColor = (this["activeBackgroundColor"] as? String)?.deserializeColor(),
        inactiveBackgroundColor = (this["inactiveBackgroundColor"] as? String)?.deserializeColor(),
        disabledBackgroundColor = (this["disabledBackgroundColor"] as? String)?.deserializeColor(),
        activeThumbColor = (this["activeThumbColor"] as? String)?.deserializeColor(),
        inactiveThumbColor = (this["inactiveThumbColor"] as? String)?.deserializeColor(),
        disabledThumbColor = (this["disabledThumbColor"] as? String)?.deserializeColor(),
    )
}
